### PR TITLE
Add clustering order support to test mock

### PR DESCRIPTION
**Pull request by mattheath**
The existing mock doesn't support clustering column clustering order, so rows returned from a timeseries (for example) will always have increasing time.

Ideally setting the clustering order for a column as a table option, such as this:

```go
thingByID = Keyspace.MultiTimeSeriesTable(
    "things_by_id",
    "ID",
    "Created", // timestamp column
    "ID",
    bucketSize,
    thing{},
).WithOptions(Options{
    ClusteringOrder: []ClusteringOrderColumn{
        ClusteringOrderColumn{
            Column:    "Created",
            Direction: DESC,
        },
    },
})
```

would ensure that the mock respected this order when returning results.

### CODE REVIEW

File: mock.go [hunk 1]
```diff
@@ -359,6 +363,17 @@ func (t *MockTable) getOrCreateColumnGroup(rowKey, superColumnKey key) map[strin
 	row := t.getOrCreateRow(rowKey)
 	scol := superColumnKey.ToSuperColumn()
 
+	// Retrieve the clustering order from the table options
+	// and assign to the key parts to set the sort ordering on each column
+	keyOrder := make(map[string]ColumnDirection, 0)
+	for _, v := range t.options.ClusteringOrder {
+		keyOrder[v.Column] = v.Direction
+		fmt.Println(v.Column, v.Direction)
```
Comment (suhailpatel) [inline mock.go:371]: Is this a stray debugging print? Probably should remove 😅

**Review by suhailpatel — DISMISSED** [review]
Approved but see comment

File: mock.go (hunk 1 above)
Comment (mattheath) [reply mock.go:371]: oh no 🤦🏻

**Review by suhailpatel — APPROVED** [review]

File: mock.go (hunk 1 above)
Comment (mattheath) [reply mock.go]: Fixed! thanks for spotting that 😅

**Pull request merged**